cmd/register/files.go

```diff
@@ -91,9 +106,23 @@ func Register(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext)
 	logger.Infof(ctx, "Parsing files... Total(%v)", len(dataRefs))
 	fastFail := !rconfig.DefaultFilesConfig.ContinueOnError
 	var registerResults []Result
+
 	for i := 0; i < len(dataRefs) && !(fastFail && _err != nil); i++ {
-		registerResults, _err = registerFile(ctx, dataRefs[i], registerResults, cmdCtx)
+		if strings.Contains(dataRefs[i], ".tar.gz") && !rconfig.DefaultFilesConfig.FastRegister {
+			return fmt.Errorf("fast serialize proto can't be register. Please use --fast flag for fast registration")
+		} else if strings.Contains(dataRefs[i], ".tar.gz") {
+			s, err := getStorageClient(ctx)
+			if err != nil {
+				return err
+			}
+			if err := uploadFastRegisterArtifact(ctx, dataRefs[i], s); err != nil {
+				return err
+			}
```

Comment:
We are just uploading for fast register and then running another command to actually register . can we combine these into one